File: api-rh/src/main/java/com/example/api_rh/model/Candidato.java
package com.example.api_rh.model;

import java.util.Date;

public class Candidato extends Pessoa {
    private String departamentoVaga;
    private Date dataInscricao;
    private Date dataEntrevista;
    private Double pretencaoSalarial;


    public Candidato() {
    }

    public Candidato(String nome, String email, String telefone, String cargo, String departamentoVaga, Double pretencaoSalarial, String cidade, String status, Date dataInscricao, Date dataEntrevista) {
        super(nome, email, telefone, cargo, cidade, status);
        this.departamentoVaga = departamentoVaga;
        this.pretencaoSalarial = pretencaoSalarial;
        this.dataInscricao = dataInscricao;
        this.dataEntrevista = dataEntrevista;
    }

    public String getDepartamentoVaga() {
        return departamentoVaga;
    }

    public void setDepartamentoVaga(String departamentoVaga) {
        this.departamentoVaga = departamentoVaga;
    }

    public Double getPretencaoSalarial() {
        return pretencaoSalarial;
    }

    public void setPretencaoSalarial(Double pretencaoSalarial) {
        this.pretencaoSalarial = pretencaoSalarial;
    }

    public Date getDataInscricao() {
        return dataInscricao;
    }

    public void setDataInscricao(Date dataInscricao) {
        this.dataInscricao = dataInscricao;
    }

    public Date getDataEntrevista() {
        return dataEntrevista;
    }

    public void setDataEntrevista(Date dataEntrevista) {
        this.dataEntrevista = dataEntrevista;
    }
}

File: web/src/types/candidato.ts
export type StatusCandidato = 'EM ANÁLISE' | 'APROVADO' | 'CONTRATADO' | 'REPROVADO' | string;

export interface Candidato {
  id: number;
  nome: string;
  email: string;
  telefone: string;
  cargo: string;
  departamentoVaga: string;
  cidade: string;
  status: StatusCandidato;
  pretencaoSalarial?: number;
  dataInscricao?: string | Date;
  dataEntrevista?: string | Date;
}

export type ModalType = 'cadastrar' | 'visualizar' | 'editar' | 'atualizar' | 'excluir' | null;

export interface CandidatoFormData {
  id?: number;
  nome: string;
  email: string;
  telefone: string;
  cargo: string;
  departamentoVaga: string;
  pretencaoSalarial?: number;
  cidade: string;
  status: string;
  dataInscricao?: string | Date;
  dataEntrevista?: string | Date;
}

export interface CandidatoPatchData {
  cargo?: string;
  status?: string;
  pretencaoSalarial?: number;
}
